### PR TITLE
Add test to verify app switcher menu in RHODS Dashboard

### PR DESCRIPTION
**Pull request by bdattoma**
Implementation of ODS-771, fetched from old PR #376 . 
I've just done some small changes

Signed-off-by: bdattoma <bdattoma@redhat.com>

### CODE REVIEW

**Comment by bdattoma** [comment]
Jenkins Job: [Job#1156](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1156/). Test cases included: ODS-771 (the new one added by this PR), ODS-327, ODS-531, ODS-1305. The other TCs have been selected because they all use a common kw which got slightly changed by the current PR.
__RUNNING__

**Comment by bdattoma** [comment]
tested on Production environment: [Job#1161](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1161/TestReport/) _SUCCESS_ the failures are expected (there's one failure due to timeout request to check an external link, but it's okay. I've checked the link and it is working)

File: tests/Resources/Common.robot [hunk 1]
```diff
@@ -143,3 +144,16 @@ Clean Resource YAML Before Creating It
     Remove From Dictionary    ${clean_yaml_data}[metadata]  managedFields  resourceVersion  uid  creationTimestamp  annotations
     [Return]   ${clean_yaml_data}
 
+Wait Until OCM Cluster Page Is Loaded
```
Comment (jgarciao) [inline tests/Resources/Common.robot:147]: I think this keyword, `Login to OCM` and `Wait OCM Splash Page` are already implemented in tests/Resources/Page/HybridCloudConsole/HCCLogin.robot

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot [hunk 2]
```diff
@@ -729,7 +738,37 @@ Check And Uninstall Operator In Openshift
     END
     Close All Browsers
 
+Check Application Switcher Links To Openshift Cluster Manager
+    [Documentation]    Checks for HTTP status of OCM link in application switcher
+    ${cluster_id}=    Get Cluster ID
+    ${cluster_id}=    Remove String    ${cluster_id}    "
+    ${ocm_staging_link}=    Set Variable    https://qaprodauth.cloud.redhat.com/openshift/details/${cluster_id}
```
Comment (jgarciao) [inline tests/Tests/400__ods_dashboard/400__ods_dashboard.robot:745]: This line could be inside of the IF statement to understand better the code

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot [hunk 3]
```diff
@@ -729,7 +738,37 @@ Check And Uninstall Operator In Openshift
     END
     Close All Browsers
 
+Check Application Switcher Links To Openshift Cluster Manager
+    [Documentation]    Checks for HTTP status of OCM link in application switcher
+    ${cluster_id}=    Get Cluster ID
+    ${cluster_id}=    Remove String    ${cluster_id}    "
```
Comment (jgarciao) [inline tests/Tests/400__ods_dashboard/400__ods_dashboard.robot:744]: Can " be removed inside `Get Cluster ID`?

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot [hunk 4]
```diff
@@ -729,7 +738,37 @@ Check And Uninstall Operator In Openshift
...
+    ${ocm_staging_link}=    Set Variable    https://qaprodauth.cloud.redhat.com/openshift/details/${cluster_id}
+    ${list_of_links}=    Get Links From Switcher
```
Comment (jgarciao) [inline tests/Tests/400__ods_dashboard/400__ods_dashboard.robot:746]: Lines 746 and 747 could be inside the ELSE statement

**Review by jgarciao — CHANGES_REQUESTED** [review]

File: tests/Resources/Common.robot (hunk 1 above)
Comment (bdattoma) [reply tests/Resources/Common.robot:147]: nope. they are different. The file you mentioned contains `Login to HCC` and `Wait For HCC Splash Page`

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot (hunk 3 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/400__ods_dashboard.robot:744]: checking this one

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot (hunk 2 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/400__ods_dashboard.robot:745]: ack

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot (hunk 4 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/400__ods_dashboard.robot:746]: ack

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot (hunk 2 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/400__ods_dashboard.robot]: done

File: tests/Tests/400__ods_dashboard/400__ods_dashboard.robot (hunk 4 above)
Comment (bdattoma) [reply tests/Tests/400__ods_dashboard/400__ods_dashboard.robot]: done

**Comment by bdattoma** [comment]
new Jenkins Job: [#1180](https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/1180/console) _SUCCESS_

File: tests/Resources/Common.robot (hunk 1 above)
Comment (bdattoma) [reply tests/Resources/Common.robot]: OCM keyword moved to a dedicated resource file

**Review by jgarciao — APPROVED** [review]

**Review by lugi0 — APPROVED** [review]
lgtm

**Pull request merged**